src/cpp/channel.cpp

```diff
@@ -63,8 +63,10 @@ StoredArray Channel::pop(const sol::optional<int>& duration,
     }
 
     auto ret = data_->channel_.front();
-    for (const auto& obj: ret)
+    for (const auto& obj: ret) {
```

Comment:
IMHO looks like workaround.
We should transform stored array to gc object.

Reply:
I don't think so. Even if `StoredArray` becomes `GCObject` we will need to revert weak ref to strong ref here. I mean it will look different but still will do the same. Isn't it?

Reply:
In fact transition from `weak ref - strong ref` can fire only when object become present at Lua state.
This taking strong refs directly from weak rers  makes logic a bit more sophisticated.
I am thinking about refactoring this..
But let it be like this.